src/game/Spell.cpp

```diff
@@ -4508,6 +4508,14 @@ SpellCastResult Spell::CheckCast(bool strict)
         {
             case SPELL_EFFECT_DUMMY:
             {
+                //By Spell ID
+                if (m_spellInfo->Id == 19938)               // Awaken Lazy Peon
+                {
+                    Unit* target = m_targets.getUnitTarget();
+                    if (!target || !target->HasAura(17743 /*SPELL_PEON_SLEEP*/) || target->GetEntry() != 10556 /*NPC_SLEEPING_PEON*/)
```

Comment:
Should consider moving the comments out of the calls, normally comments are only done directly in code in declarations, in calls we comment above usually, the comment could looks something like
`// 17743 = Lazy Peon Sleep | 10556 = Lazy Peon`